dot/build_spec_test.go

```diff
@@ -0,0 +1,289 @@
+    "bootNodes": null,
+    "telemetryEndpoints": null,
+    "protocolId": "protocol",
+    "genesis": {},
+    "properties": null,
+    "forkBlocks": null,
+    "badBlocks": null,
+    "consensusEngine": "",
+    "codeSubstitutes": null
+}`,
+		},
+		{
+			name: "normal conditions",
+			fields: fields{genesis: &genesis.Genesis{
+				Name:               "test",
+				ID:                 "ID",
+				ChainType:          "chainType",
+				Bootnodes:          []string{"node1", "node2"},
+				TelemetryEndpoints: []interface{}{"endpoint"},
+				ProtocolID:         "protocol",
+				Genesis:            genesis.Fields{},
+				Properties:         map[string]interface{}{"key": "value"},
+				ForkBlocks:         []string{"1", "2"},
+				BadBlocks:          []string{"3", "4"},
+				ConsensusEngine:    "babe",
+				CodeSubstitutes:    map[string]string{"key": "value"},
+			}},
+			want: `{
+    "name": "test",
+    "id": "ID",
+    "chainType": "chainType",
+    "bootNodes": [
+        "node1",
+        "node2"
+    ],
+    "telemetryEndpoints": null,
+    "protocolId": "protocol",
+    "genesis": {},
+    "properties": {
+        "key": "value"
+    },
+    "forkBlocks": null,
+    "badBlocks": null,
+    "consensusEngine": "",
+    "codeSubstitutes": null
+}`,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b := &BuildSpec{
+				genesis: tt.fields.genesis,
+			}
+			got, err := b.ToJSON()
+			if tt.err != nil {
+				assert.ErrorIs(t, err, tt.err)
+			} else {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, tt.want, string(got))
+		})
+	}
+}
+
+func TestBuildFromDB(t *testing.T) {
+	// initialise node (initialise state database and load genesis data)
+	cfg := NewTestConfig(t)
+	cfg.Init.Genesis = "../chain/gssmr/genesis.json"
+	nodeInstance := nodeBuilder{}
+	err := nodeInstance.initNode(cfg)
+	assert.NoError(t, err)
+
+	basePath := t.TempDir()
+	expectedPath := basePath[:len(basePath)-1] + "1"
+
+	type args struct {
+		path string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *BuildSpec
+		err  error
+	}{
+		{name: "normal conditions", args: args{path: expectedPath},
+			want: &BuildSpec{genesis: &genesis.Genesis{Name: "Gossamer"}}},
+		{name: "invalid db path", args: args{path: "foo/bar"},
+			err: errors.New("cannot start state service: failed to create block state: cannot get block 0: Key not found")},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := BuildFromDB(tt.args.path)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+			if tt.want != nil {
+				assert.Equal(t, tt.want.genesis.Name, got.genesis.Name)
+			}
+			// remove files created for tests
+			err = os.RemoveAll(tt.args.path)
+			assert.NoError(t, err)
+		})
+	}
+}
+
+func TestBuildFromGenesis(t *testing.T) {
+	// setup test file
+	file := genesis.CreateTestGenesisJSONFile(t, false)
+	defer os.Remove(file)
```

Comment:
I don't this is needed right, `genesis.CreateTestGenesisJSONFile` should take care of removing it with `t.TempDir()`

```suggestion
```